staging/src/k8s.io/apiserver/pkg/authentication/request/x509/x509_test.go

```diff
@@ -898,7 +873,12 @@ func TestX509Verifier(t *testing.T) {
 			return &authenticator.Response{User: &user.DefaultInfo{Name: "innerauth"}}, true, nil
 		})
 
-		a := NewVerifier(testCase.Opts, auth, testCase.AllowedCNs)
+		a := &Verifier{
+			rootsFn:            staticRootFn(testCase.Roots),
+			clock:              clocktesting.NewFakePassiveClock(testCase.CurrentTime),
+			auth:               auth,
+			allowedCommonNames: StaticStringSlice(testCase.AllowedCNs.List()),
+		}
```

Comment:
Can we keep using the `New...` function and just override the fields we need to?  It is fine to cast to the struct if needed.